src/main/java/com/riwi/localstorage/riwi_local_storage/infrastructure/services/StoreService.java

```diff
@@ -73,15 +75,25 @@ public StoreResponse update(String id, StoreRequestUpdate request) {
         throw new UnsupportedOperationException("Unimplemented method 'update'");
     }
 
+    /*----------------------
+     * DISABLE ROLE (SOFT DELETE)
+     * ---------------------
+     */
+
     @Override
-    public void delete(String id) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'delete'");
+    public void delete(Integer id) {
+        Store storeDisable = this.find(id);
+        if (storeDisable.getStatus() == StatusType.INACTIVE) {
+            throw new RoleAlreadyInactiveException("Store with ID " + id + " is already inactive.");
```

Comment:
Modifies this Exception `(RoleAlreadyInactiveException)` . You must create the store exception and not use the role exception.

Reply:
I've already create the store exception and commited the changes.